R/MoBiProject.R

```diff
@@ -119,42 +193,112 @@ MoBiProject <- R6::R6Class(
-      modules <- self$getModules(modulesNames)
-      individual <- self$getIndividual(individualName, stopIfNotFound = FALSE)
-      expressionProfiles <- self$getExpressionProfiles(expressionProfilesNames, stopIfNotFound = FALSE)
-
-      configuration <- createSimulationConfiguration(
-        modules = modules,
-        individual = individual,
-        expressionProfiles = expressionProfiles,
-        selectedInitialConditions = selectedInitialConditions,
-        selectedParameterValues = selectedParameterValues
+      # Task used for getting information (IC and PV BBs) from modules
+      modulesTask <- .getNetTaskFromCache("ModuleTask", isMoBiR = TRUE)
+
+      modules <- self$getModules()
+      # Throw an error when a specified module is not present in the project
+      missingModules <- setdiff(modulesNames, names(modules))
+      if (length(missingModules) > 0) {
+        stop(messages$modulesNotPresentInProject(missingModules))
+      }
+
+      # Get the specified individual
+      individual <- NULL
+      if (!is.null(individualName)) {
+        individual <- self$getIndividual(individualName, stopIfNotFound = TRUE)
+      }
+      # Get the specified expression profiles
+      expressionProfiles <- list()
+      if (!is.null(expressionProfilesNames)) {
+        expressionProfiles <- self$getExpressionProfiles(expressionProfilesNames, stopIfNotFound = TRUE)
+      }
+
+      # Create module configurations for each module
+      moduleConfigurations <- lapply(modulesNames, function(moduleName) {
+        module <- modules[[moduleName]]
+
+        icBB <- NULL
+        # If no IC BB is specified, use the first IC BB available in the module
+        if (is.null(selectedInitialConditions)) {
+          # First get the list of all IC BBs
+          icBBs <- modulesTask$call("AllInitialConditionsFromModule", module)
+          icBBs <- icBBs$call("ToArray")
+          # Select the first IC BB if available
+          if (length(icBBs) > 0) {
+            icBB <- icBBs[[1]]
+          }
+        } else {
+          # If initial conditions selection have been provided, use them.
+          # If the name of the module is in the names of 'selectedInitialConditions',
+          # get the value. The value could be NULL, of no IC BB should be selected.
+          if (moduleName %in% names(selectedInitialConditions)) {
+            selectedICName <- selectedInitialConditions[[moduleName]]
+            icBB <- modulesTask$call("InitialConditionBuildingBlockByName", module, selectedICName)
+            # However, if the provided value for the name is NULL, use NULL to specify no IC BB
+            # Cannot create configuration if the speciefied IC BB is not available
+            if (is.null(icBB) && !is.null(selectedICName)) {
+              stop(messages$icBBNotPresentInModule(moduleName, selectedICName))
+            }
```

Comment:
would modify this to
```
           selectedICName <- selectedInitialConditions[[moduleName]]
           if (!is.null(selectedICName)) {
              icBB <- modulesTask$call("InitialConditionBuildingBlockByName", module, selectedICName)
              # Cannot create configuration if the speciefied IC BB is not available
              if (is.null(icBB)) {
                stop(messages$icBBNotPresentInModule(moduleName, selectedICName))
              }
            }
```

Reply:
It is possible that `selectedICName`, if the user wants that NO IC is selected. In this case, `icBB <- modulesTask$call("InitialConditionBuildingBlockByName", module, selectedICName)` will also evaluate to `NULL` and later passed to `.createModuleConfiguration` call.

I don't think this behavior is possible with your proposal?